modules/apps/27-interchain-accounts/host/keeper/relay_test.go

```diff
@@ -448,35 +450,224 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
+
+				params := types.NewParams(true, []string{sdk.MsgTypeURL((*banktypes.MsgSend)(nil))})
+				suite.chainB.GetSimApp().ICAHostKeeper.SetParams(suite.chainB.GetContext(), params)
+			},
+			false,
+		},
+	}
+
 	for _, tc := range testCases {
 		tc := tc
 
 		suite.Run(tc.msg, func() {
 			suite.SetupTest() // reset
 
-			path = NewICAPath(suite.chainA, suite.chainB)
+			path = NewICAPath(suite.chainA, suite.chainB, icatypes.EncodingJSON)
 			suite.coordinator.SetupConnections(path)
 
-			err := SetupICAPath(path, TestOwnerAddress)
+			err := SetupICAPath(path, TestOwnerAddress, icatypes.EncodingJSON)
 			suite.Require().NoError(err)
 
 			portID, err := icatypes.NewControllerPortID(TestOwnerAddress)
 			suite.Require().NoError(err)
 
-			// Get the address of the interchain account stored in state during handshake step
-			storedAddr, found := suite.chainB.GetSimApp().ICAHostKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), ibctesting.FirstConnectionID, portID)
-			suite.Require().True(found)
-
-			icaAddr, err := sdk.AccAddressFromBech32(storedAddr)
-			suite.Require().NoError(err)
-
-			// Check if account is created
-			interchainAccount := suite.chainB.GetSimApp().AccountKeeper.GetAccount(suite.chainB.GetContext(), icaAddr)
-			suite.Require().Equal(interchainAccount.GetAddress().String(), storedAddr)
+			// Set the address of the interchain account stored in state during handshake step for cosmwasm testing
+			suite.setICAWallet(suite.chainB.GetContext(), portID, interchainAccountAddr)
```

Comment:
Just for my understanding: so why do you need to set the interchain account address yourself and cannot instead use the code that was here before to retrieve with `GetInterchainAccountAddress`?

Reply:
Because there are pre-serialized bytes used in the test, for example, see line 641

Reply:
I've fixed this issue my making tests human readable and moving to JSON tests instead of cosmwasm tests